Reformatting misc and newcat moon events

CL0WNTH0UGHTS opened the pull request:
Finishes reformatting all misc and new_cat events to the new format, fixes certain issues such as 
-kittens being confessed to murders
-making new_cat litter events generate a outsider

Some notes:

-used "clan" in relationship tags, I understand if it doesn't work and will be an easy patch if it ends up not working out
-No idea if my weights are balanced, just did what seemed right
-Added a 'test' event to see if new_cats can generate a polycat to join a couple. If it crashes, it's ID is gen_new_cat_anypolymate1

CL0WNTH0UGHTS committed:
Reformat beach.json

Reformats all four beach-specific moon events to the new format. Here are some notes;

-allows medicine cat apprentices to get omens/prophet lists (I have no good reason for doing so, I just think players could build some good stories due to an apprentice being sent something from StarClan)

-make the med cat list events fairly uncommon (can change it to rare, my personal weight is 5 - 10)

-made the leader event need a skill (I don't know if this will limit it harder, feel free to tell me to remove it)

rest should be fairly explainable, feel free to bonk me and ill try to give a reason :3.

CL0WNTH0UGHTS committed:
Reformat forest.json

Reformats all three misc events for the forest. Same notes as beach.json (made it so medicine cat apprentices can get the omen/prophet moon events).

CL0WNTH0UGHTS committed:
Minor spelling mistake

I misspelled the clan_apps tag, forgive me :pray:

CL0WNTH0UGHTS committed:
minor spelling mistake

misspelled the tag "clan_apps", just fixes it

CL0WNTH0UGHTS committed:
Reformat Mountainous.json

reformat all mountainous misc events, usual notes about the events.

CL0WNTH0UGHTS committed:
Update/fix skills

This basically just changes the omen/prophet from 1, to 2-3. This also fixes an omen event having a prophet skill trait requirement

CL0WNTH0UGHTS committed:
Reformat plains.json

Reformats all plains.json misc events

CL0WNTH0UGHTS committed:
reformat general.json

This reformats all general.json events, aswell as semi-organizes them.

CL0WNTH0UGHTS committed:
Update misc events general.json

-Adds "clan_dsicovery tag" to events that either mention a Clan or don't mention a specific other cat

-Add some status constraints for certain mate events that mention patrolling (so that elder mates can't get it)

-Adds romance tag to events that are not defined as mate events, but increase romance

CL0WNTH0UGHTS committed:
Update + add new_cat general.json

Reformats all new_cat general.json events to the format

Add gen_new_cat_anypolymate1 as an experimental new_cat event (generates a new_cat to be the mate of m_c and r_c)

scribblecrumb commented on resources/dicts/events/misc_events/beach.json:
```diff
@@ -1,118 +1,53 @@
+    "biome": ["beach"],
+    "camp": ["any"],
+    "season": ["any"],
+    "tags": ["war", "clan_apps"],
```
I assume you've already fixed subtypes and just haven't updated this PR?

scribblecrumb commented on resources/dicts/events/misc_events/beach.json:
```diff
@@ -1,118 +1,53 @@
+    "event_text": "As a storm rages across the beach and waves crash upon the shore, m_c is tucked up in the medicine cat den, {PRONOUN/m_c/poss} sleep anything but restful. {PRONOUN/m_c/subject/CAP} {VERB/m_c/keep/keeps} seeing flashes of prophecy_list_sight. {PRONOUN/m_c/subject/CAP} {VERB/m_c/wake/wakes} with a shiver, wondering what {PRONOUN/m_c/poss} dreams mean for the Clan.",
+    "m_c": {
+      "status": ["medicine cat", "medicine cat apprentice"],
+      "skill": ["PROPHET,2", "PROPHET,3"]
```
likewise, do you already plan to adjust these now that it's been cleared up how skill constraints work?

CL0WNTH0UGHTS replied:
Have done so, aswell as many other events that have skill parameters. Thank you for notifying me :3

scribblecrumb commented on resources/dicts/events/misc_events/beach.json:
```diff
@@ -1,118 +1,53 @@
+    "weight": 15,
+    "event_text": "The apprentices of the Clan are taking special training to learn how to disguise their scents from enemies with kelp and seawater.",
+    "m_c": {
+      "status": ["apprentice"]
```
please also include `age` with "any".  both `age` and `status` params should be included with "any" if they would otherwise be empty

CL0WNTH0UGHTS replied:
Should also hopefully be fixed, updated all misc events across the board with the parameters. Ill update the new_cat events shortly (not marking as resolved to remind myself)

scribblecrumb commented on resources/dicts/events/misc_events/forest.json:
```diff
@@ -1,89 +1,38 @@
+    "event_id": "fst_misc_multi_prophecymedcatprophet1",
+    "biome": ["forest"],
+    "camp": ["any"],
+    "season": ["Newleaf", "Greenleaf", "Leaf-fall"],
```
let's make these lowercase

CL0WNTH0UGHTS replied:
Fixed, thank you for pointing that out

scribblecrumb commented on resources/dicts/events/misc_events/general.json:
```diff
@@ -1,5659 +1,2653 @@
+    "trait": ["bloodthirsty"]
+  },
+  "r_c": {
+    "age": ["kitten","adolescent", "young adult", "adult", "senior adult", "senior"]
```
let's remove the kitten and adolescent to help avoid cats who aren't old enough to have witnessed the murder from getting this event.  it'll still be possible, but hopefully not as likely.  i hope eventually we can add some sort of filter to help prevent this in general, but for now this is easier.

CL0WNTH0UGHTS replied:
Changed within [86bfd96](https://github.com/scribblecrumb/clangen/pull/27/commits/86bfd9648c8800dc11b2893afc7b55e64701738a), aswell as other events in the same vein

CL0WNTH0UGHTS committed:
Update misc_events/plains.json

-Adds the "sub_type" parameter to the events that required them (pln_misc_war_anyapps1)
-Updates the skills to reflect the actual encoding, maintains the skills being at the second level, as prophet1 and omen1 do not make sense but omen4 and prophet4 do (pln_misc_anyprophecymedcatprophet1 & pln_misc_anyomenmedcatomen1)

CL0WNTH0UGHTS committed:
Update misc_events\mountanious.json

-Adds the "sub_type" parameter to the events that required them (mtn_misc_war_anyapps1)
-Updates the skills to reflect the actual encoding, maintains the skills being at the second level, as prophet1 and omen1 do not make sense but omen4 and prophet4 do (mtn_misc_anyomenmedcatomen1 & mtn_misc_anyprophecymedcatprophet1)
-Update the "age" parameter (as I was mistaken about it being not needed; it is, in fact, needed)

CL0WNTH0UGHTS committed:
Update misc_events\forest.json

-Adds the "sub_type" parameter to the events that required them (fst_misc_war_anyapps1)
-Updates the skills to reflect the actual encoding, maintains the skills being at the second level, as prophet1 and omen1 do not make sense but omen4 and prophet4 do (fst_misc_multiprophecymedcatprophet1 & fst_misc_multiomenmedcatomen1)
-Update the "age" parameter (as I was mistaken about it being not needed; it is, in fact, needed)

CL0WNTH0UGHTS committed:
Update misc_events\beach.json

-Updates some events IDs
-Add "sub_type" parameter to events that needed it
-Add "age" parameter to all events

CL0WNTH0UGHTS committed:
Update misc_events\general.json

-Added either/both "age" or "status" parameter
 -Likewise, removed certain parameters

-Add "sub_class" and update events accordingly

CL0WNTH0UGHTS committed:
Update new_cat\general.json

-Adds the age and status parameter to any events that required it

-fix some repeating events

CL0WNTH0UGHTS committed:
Update new_cat\general.json

adds the parameter  "sub_types" to events who needed it (adoption and war events)

scribblecrumb commented on resources/dicts/events/new_cat/general.json:
```diff
@@ -4,11 +4,12 @@
         "camp": ["any"],
         "season": ["any"],
-        "tags": ["adoption"],
+        "sub_types": ["adoption"],
```
make sure it's `sub_type` not `sub_types`

CL0WNTH0UGHTS replied:
Fixed in b7da8fe, thank you for notifying me

scribblecrumb commented on resources/dicts/events/new_cat/general.json:
```diff
@@ -411,6 +426,10 @@
         "season": ["any"],
         "weight": 20,
         "event_text": "m_c finds a rogue named n_c_pre, who joins the Clan. The rogue decides to take on a more Clan-like name, and is now called n_c.",
```
n_c abbrs need to follow the abbrs that patrols have been using. that is, the `n_c:{index}` system.  n_c_pre can be converted to `n_c_pre:{index]`

CL0WNTH0UGHTS committed:
Updare injury\beach.json

Reformats all beach specific injury events to the new formatting

CL0WNTH0UGHTS committed:
Update plains injury json

Updates all injury events found within injury\plains.json to the new formatting

CL0WNTH0UGHTS committed:
Update mountainous injury json

Updates all injury events found within injury\mountanious.json to the new formatting

CL0WNTH0UGHTS committed:
Update general injury events

Updates all injury events found within injury\general.json to the new formatting

CL0WNTH0UGHTS committed:
Update forest injury events

Updates all injury events found within injury\forest.json to the new formatting

CL0WNTH0UGHTS committed:
Fix minor grammar issue

Replace all instances of sub_types to sub_type

CL0WNTH0UGHTS committed:
Fixes abbreviation error

Corrects all n_c instances to the index version that is used (Such as replacing n_c_pre to n_c_pre:0)
